libbeat/processors/rate_limit/rate_limit.go

```diff
@@ -101,26 +103,30 @@ func (p *rateLimit) String() string {
 	)
 }
 
-func (p *rateLimit) makeKey(event *beat.Event) (string, error) {
-	var key string
-
-	if len(p.config.Fields) > 0 {
-		sort.Strings(p.config.Fields)
-		values := make([]string, len(p.config.Fields))
-		for _, field := range p.config.Fields {
-			value, err := event.GetValue(field)
-			if err != nil && err == common.ErrKeyNotFound {
-				value = ""
-			}
-			if err != nil {
-				return "", errors.Wrapf(err, "error getting value of field: %v", field)
-			}
-
-			// TODO: check that the value is a scalar?
-			values = append(values, fmt.Sprintf("%v", value))
+func (p *rateLimit) makeKey(event *beat.Event) (uint64, error) {
+	if len(p.config.Fields) == 0 {
+		return 0, nil
+	}
+
+	sort.Strings(p.config.Fields)
+	values := make([]string, len(p.config.Fields))
+	for _, field := range p.config.Fields {
+		value, err := event.GetValue(field)
+		if err != nil && err != common.ErrKeyNotFound {
+			return 0, errors.Wrapf(err, "error getting value of field: %v", field)
+		}
+		if err != common.ErrKeyNotFound {
+			value = ""
```

Comment:
Is this setting `value` to the empty string if `err == nil`?

Reply:
Yes, it is 🤦 . It needs an extra `err != nil` check. I will add it. Thanks for catching!

Reply:
Fixed in 8a758888a.